### PR TITLE
Display correct error when file upload and add more description

### PR DESCRIPTION
**Pull request by samchan0221**
ref #1065

### CODE REVIEW

File: pkg/api/apierrors/kinds.go [hunk 1]
```diff
@@ -50,3 +53,5 @@ func (n Name) WithReason(reason string) Kind {
 }
 
 var ValidationFailed = Invalid.WithReason("ValidationFailed")
+
+var FileUploadTooLarge = RequestEntityTooLarge.WithReason("FileUploadTooLarge")
```
Comment (louischan-oursky) [inline pkg/api/apierrors/kinds.go:57]: Name the Reason as `ResourceTooLarge`.

```go
var ResourceTooLarge = apierrors.RequestEntityTooLarge.WithReason("ResourceTooLarge")
```

Define this in `pkg/portal/appresource/errors.go`. This file does not exist yet so please create it.

File: pkg/portal/appresource/manager.go [hunk 2]
```diff
@@ -109,7 +109,9 @@ func (m *Manager) ApplyUpdates(appID string, updates []Update) ([]*resource.Reso
 	// Validate file size.
 	for _, f := range updates {
 		if len(f.Data) > ConfigFileMaxSize {
-			return nil, fmt.Errorf("invalid resource '%s': too large (%v > %v)", f.Path, len(f.Data), ConfigFileMaxSize)
+			message:= fmt.Sprintf("invalid resource '%s': too large (%v > %v)", f.Path, len(f.Data), ConfigFileMaxSize)
+			err := apierrors.FileUploadTooLarge.NewWithInfo(message, apierrors.Details{"requestSize": len(f.Data), "maxSize": ConfigFileMaxSize})
```
Comment (louischan-oursky) [inline pkg/portal/appresource/manager.go:113]: Our convention is snake_case. So the details dictionary should be

```golang
apierrors.Details{
  "size": len(f.Data),
  "max_size": ConfigFileMaxSize,
  // Also include path
  "path": f.Path,
}
```

File: portal/src/error/fileUplaod.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export interface APIFileUploadTooLargeError {
```
Comment (louischan-oursky) [inline portal/src/error/fileUplaod.ts:1]: Put this in `portal/src/error/resources.ts`. Remember the name and the reason should have changed.

**Review by louischan-oursky — COMMENTED** [review]
This comment https://github.com/authgear/authgear-server/issues/1065#issuecomment-892359883 requires us to mention the "name" of the resource in the error message. See if the information provided by `Path` is enough.

**Comment by samchan0221** [comment]
Updated corresponding to new requests from PM and reviewer

File: portal/src/locale-data/en.json [hunk 4]
```diff
@@ -643,7 +644,7 @@
   "ThemeConfigurationWidget.dark-mode": "Dark Mode",
   "ThemeConfigurationWidget.theme-color-title": "Theme Color",
   "ThemeConfigurationWidget.app-logo-title": "App Logo",
-  "ThemeConfigurationWidget.app-logo-description": "It helps the end users to identify your app, and it is shown in all pages, emails, etc. JPEG, PNG and GIF are supported.",
+  "ThemeConfigurationWidget.app-logo-description": "It helps the end users to identify your app, and it is shown in all pages, emails, etc. JPEG, PNG and GIF are supported. \n (Size limit: 100 KB)",
```
Comment (louischan-oursky) [inline portal/src/locale-data/en.json:647]: You can `use {br, react}` to insert a `<br>` element. So we do not need `white-space: pre-line;`

**Comment by louischan-oursky** [comment]
Build failed.

**Pull request merged**